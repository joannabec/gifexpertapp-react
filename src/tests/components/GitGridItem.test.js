/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

const title = 'Titulo';
const url = 'https://algunsitio.com/foto.jpg';
const wrapper = shallow(<GifGridItem title={title} url={url} />);

describe('Pruebas del componente <GifGridItem/>', () => {
  test('Debería renderizar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debería tener un párrafo con el título', () => {
    const paragraph = wrapper.find('p');
    expect(paragraph.text().trim()).toBe(title);
  });

  test('Deberían coincidir los atributos src y alt con el url y title', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Debería tener la clase correcta', () => {
    const div = wrapper.find('div');
    expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
  });
});
