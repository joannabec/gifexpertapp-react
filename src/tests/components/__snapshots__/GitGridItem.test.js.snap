// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGridItem/> Deber√≠a renderizar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://algunsitio.com/foto.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
